modules/plugin/jdbc/jdbc-sqlserver/src/main/java/org/geotools/data/sqlserver/SQLServerDialect.java

```diff
@@ -1042,4 +1043,52 @@ public Object getNextSequenceValue(String schemaName, String sequenceName, Conne
             return null;
         }
     }
+
+    @Override
+    public List<ReferencedEnvelope> getOptimizedBounds(
+            String schema, SimpleFeatureType featureType, Connection cx) throws SQLException {
+
+        if (null != dataStore.getVirtualTables().get(featureType.getTypeName())) {
+            return null;
+        }
+
+        // select the x/y of opposing corners of the aggregate envelope
+        final StringBuffer sql = new StringBuffer("SELECT geometry::EnvelopeAggregate(");
+        encodeColumnName(null, featureType.getGeometryDescriptor().getLocalName(), sql);
+        sql.append(").STPointN(1).STX, geometry::EnvelopeAggregate(");
+        encodeColumnName(null, featureType.getGeometryDescriptor().getLocalName(), sql);
+        sql.append(").STPointN(1).STY, geometry::EnvelopeAggregate(");
+        encodeColumnName(null, featureType.getGeometryDescriptor().getLocalName(), sql);
+        sql.append(").STPointN(3).STX, geometry::EnvelopeAggregate(");
+        encodeColumnName(null, featureType.getGeometryDescriptor().getLocalName(), sql);
+        sql.append(").STPointN(3).STY FROM ");
+        encodeTableName(featureType.getTypeName(), sql);
```

Comment:
Just reading the text, it would seem SQLServer might end up computing the aggregate four times. I guess you verified it's not the case?
The alternative would be to read the geometry produced by EnvelopeAggregate, the dialect is able to do so anyways.

Reply:
will need to verify, tied up right now so I will mark this as draft